.github/workflows/code-style.yaml

```diff
@@ -0,0 +1,23 @@
+# Run code style checks on each pull request.
+name: Code style checks
+
+on: [pull_request]
+
+jobs:
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Pixi Environment
+        uses: prefix-dev/setup-pixi@v0.8.1
+        with:
+          pixi-version: v0.25.0
+          cache: true
+          # Only write the cache on push to main to avoid exceeding GitHub's 10GB cache limit.
+          cache-write: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
+
+      - name: Set up Python
+        run: pixi global install python
```

Comment:
This seems rather odd to me. Like we're stuck between two sensible alternatives.

If we just need Python to support the pre-commit action then we should probably just use the standard Setup-Python action.

If we want to call pre-commit ourselves via pixi run, then we should use a (likely default) environment.

In neither case would I use `global`.

I don't think we should fix this now. In particular I need to remember why I don't like the pre-commit action and we can do that in a follow-up.